tests/upgrades/test_errata.py

```diff
@@ -0,0 +1,243 @@
+        environment = entities.LifecycleEnvironment(
+            organization=org
+        ).search(query={'search': 'name=Library'})[0]
+
+        product = entities.Product(organization=org).create()
+        custom_yum_repo = entities.Repository(product=product,
+                                              content_type='yum',
+                                              url=FAKE_9_YUM_REPO
+                                              ).create()
+
+        tools_repo, rhel_repo = self._create_custom_rhel_tools_repos(client_os, product)
+        product.sync()
+        content_view = entities.ContentView(organization=org).create()
+        content_view.repository = [custom_yum_repo, tools_repo, rhel_repo]
+        content_view = content_view.update(['repository'])
+        content_view.publish()
+        content_view = content_view.read()
+        ak = entities.ActivationKey(
+            content_view=content_view,
+            organization=org.id,
+            environment=environment
+        ).create()
+        subscription = entities.Subscription(organization=org).search(query={
+            'search': 'name={}'.format(product.name)
+        })[0]
+
+        ak.add_subscriptions(data={
+            'subscription_id': subscription.id})
+        rhel7_client = dockerize(
+            ak_name=ak.name, distro='rhel7', org_label=org.label)
+        client_container_id = list(rhel7_client.values())[0]
+        self._install_package(client_container_id, 'katello-agent')
+        self._run_goferd(client_container_id)
+
+        for package in FAKE_9_YUM_OUTDATED_PACKAGES:
+            self._install_package(client_container_id, package)
+        host = entities.Host().search(query={
+            'search': 'activation_key={0}'.format(ak.name)})[0]
+        host = host.read()
+        applicable_errata_count = host.content_facet_attributes[
+            'errata_counts']['total']
+        self.assertGreater(applicable_errata_count, 1)
+
+        erratum_list = entities.Errata(repository=custom_yum_repo).search(query={
+            'order': 'updated ASC',
+            'per_page': 1000,
+        })
+        errata_ids = [errata.errata_id for errata in erratum_list]
+        self.assertEqual(sorted(errata_ids), sorted(FAKE_9_YUM_ERRATUM))
+        scenario_dict = {self.__class__.__name__: {
+            'rhel_client': rhel7_client,
+            'activation_key': ak.name,
+            'custom_repo_id': custom_yum_repo.id
+        }}
+        create_dict(scenario_dict)
+
+    @post_upgrade
+    def test_post_scenario_errata_count_installtion(self):
+        """Post-upgrade scenario that installs the package on pre-upgrade
+        client remotely and then verifies if the package installed.
+
+        :id: 88fd28e6-b4df-46c0-91d6-784859fd1c21
+
+        :steps:
+
+            1. Recovered pre_upgrade data for post_upgrade verification
+            2. Verifying errata count has not changed on satellite
+            3. Run goferd
```

Comment:
@omkarkhatavkar One step is also require to add which is `upgrade katello-agent/goferd` during post-upgrade scenario.
above is required because when Satellite upgrade to major version then content/client host **kattelo-agent** also requires to upgrade so that clients are fully compatible with Satellite Server.

Reply:
Done !

Reply:
@omkarkhatavkar thanks , please update the doc string too under post-upgrade as well.

Reply:
Done !

Reply:
@vijay8451 the latest result with updating location in host. 
```

PASSED2019-02-22 16:33:54 - robottelo - INFO - Started tearDownClass: tests.upgrades.test_errata/Scenario_errata_count


=========================================================== warnings summary ===========================================================
tests/upgrades/test_errata.py::Scenario_errata_count::test_pre_scenario_generate_errata_for_client
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
    InsecureRequestWarning)
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/bugzilla/base.py:114: DeprecationWarning: The SafeConfigParser class has been renamed to ConfigParser in Python 3.2. This alias will be removed in future versions. Use ConfigParser directly instead.
    self.tokenfile = SafeConfigParser()
  /home/okhatavk/Satellite/robottelo/env/lib/python3.4/site-packages/bugzilla/base.py:558: DeprecationWarning: The SafeConfigParser class has been renamed to ConfigParser in Python 3.2. This alias will be removed in future versions. Use ConfigParser directly instead.
    c = SafeConfigParser()

-- Docs: https://docs.pytest.org/en/latest/warnings.html
======================================== 1 passed, 29 deselected, 3 warnings in 1154.48 seconds ========================================

```